src/evalUtils.js

```diff
@@ -16,10 +16,28 @@ const logger = require('winston');
 const sampleSchema = require('./schema').sample;
 const commonUtils = require('./common');
 const { VM } = require('vm2');
-const template = require('just-template');
 
 const EVAL_TIMEOUT_MILLIS = 750;
 
+function template(str, data) {
+  const keysToReplaceRegex = /\{\{([^\}]+)?\}\}/g;
+  const arrayKeyRegEx = /([\\$\w\d_-]+)\[(\d+)\]/;
+  return str.replace(keysToReplaceRegex, (_, key) => {
+    const keyParts = key.split('.');
+    let value = data;
+    keyParts.forEach((k) => {
+      const arrayKey = k.match(arrayKeyRegEx);
+      if (arrayKey) {
+        value = value[arrayKey[1]][arrayKey[2]];
+      } else {
+        value = value[k];
+      }
+    });
+
+    return value || '';
+  });
+} // template
```

Comment:
Might be outside the scope of this story since it wasn't previously supported by justTemplate and it's not as common a use case, but worth thinking about for future improvements: Would it be useful for this to also support "xxx[yyy]"?

I could see a situation where you want to access a different property depending on  the subject/aspect, or another context var. For example, if you want to map subject names to superpods to build the url: "https://www.example.com/sp={{subSpMap[subject.name]}}".

Of course, you can always just use toUrl instead, but it might be worth supporting this for convenience.

I think for now we should at least add tests for this case to make it clear that it's not currently supported.

Also, a similar but probably less common case would be a multi-dimensional array. ("arr[2][0]")

Reply:
Let's not go down that rabbit hole right now... but if you want to create a user story for it, go for it.